apps/OpenSpace/ext/launcher/include/sgctedit/monitorbox.h

```diff
@@ -0,0 +1,95 @@
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_UI_LAUNCHER___MONITORBOX___H__
+#define __OPENSPACE_UI_LAUNCHER___MONITORBOX___H__
+
+#include <QColor>
+#include <QIntValidator>
+#include <QLineEdit>
+#include <QPainter>
+#include <QPainterPath>
+#include <QPoint>
+#include <QVector>
+#include <QWidget>
+
+#include <algorithm>
+#include <vector>
+#include <iostream>
+
+#include "windowcontrol.h"
+
+
+class MonitorBox : public QWidget
+{
+Q_OBJECT
+public:
+    explicit MonitorBox(QRect widgetDims, std::vector<QRect> monitorResolution,
+        unsigned int nWindows, QString* winColors);
+    ~MonitorBox();
+    void mapMonitorResolutionToWidgetCoordinates();
+    void mapWindowResolutionToWidgetCoordinates(unsigned int mIdx, unsigned int wIdx,
+        const QRectF& w);
+    void setNumWindowsDisplayed(unsigned int nWindows);
+    void windowDimensionsChanged(unsigned int monitorIdx, unsigned int windowIdx,
+            const QRectF& newDimensions);
+
+protected:
+    void paintEvent(QPaintEvent *event) override;
+
+private:
+    void paintWidgetBorder(QPainter& painter, const int width, const int height);
+    void paintMonitorBackgrounds(QPainter& painter);
+    void paintWindow(QPainter& painter, const unsigned int winIdx);
+    void paintWindowBeyondBounds(QPainter& painter, const unsigned int winIdx);
+    void paintWindowNumber(QPainter& painter, const unsigned int winIdx);
+    void setPenSpecificToWindow(QPainter& painter, const unsigned int windowIdx,
+        bool visibleBorder);
+    void computeScaledResolution_landscape(const float aspectRatio, const float maxWidth);
+    void computeScaledResolution_portrait(const float aspectRatio, const float maxHeight);
+    unsigned int _maxNumMonitors = 2;
+    QRectF _monitorWidgetSize;
+    QRectF _monitorBoundaryRect;
+    unsigned int _nMonitors = 1;
+    std::vector<QRect> _monitorResolution;
+    std::vector<QRectF> _monitorDimensionsScaled;
+    std::vector<QRectF> _windowResolutions;
+    std::vector<QRectF> _windowRendering = {
+        {0.0, 0.0, 0.0, 0.0},
+        {0.0, 0.0, 0.0, 0.0},
+        {0.0, 0.0, 0.0, 0.0},
+        {0.0, 0.0, 0.0, 0.0}
+    };
+    unsigned int _nWindows = 1;
+    QString* _colorsForWindows;
+    int _alphaWindowOpacity = 170;
+    float _monitorScaleFactor = 1.0;
+    float _offset[2] = {10.0, 10.0};
+    bool _showLabel = false;
+    float _marginFractionOfWidgetSize = 0.025;
```

Comment:
This is not actually a variable and should be moved into the `cpp` file and declared as a `constexpr const MarginFractionOfWidgetSize` since it is a compile-time setting instead

Reply:
I think you mean `constexpr float`

Reply:
Eah.. jap 😅